misc/python/materialize/cli/mzcompose.py

```diff
@@ -710,6 +713,178 @@ def write_junit_report_to_file(self, junit_suite: junit_xml.TestSuite) -> Path:
+                )
+
+            detected_commit = self.run_bisect(
+                args,
+                older_commit_hash=oldest_commit_hash,
+                newer_commit_hash=newest_commit_hash,
+                older_commit_success=oldest_commit_success,
+                newer_commit_success=newest_commit_success,
+                tag_param_instead_checkout=tag_param_instead_checkout,
+                error_message_must_contain=error_message_must_contain,
+                log=log,
+            )
+            log.append(f"Detected commit: {detected_commit}")
+
+            print("\n".join(log))
+        finally:
+            if change_git_revision:
+                # go back to the original state
+                git.checkout(original_revision)
+
+    def validate_args(self, args: argparse.Namespace) -> None:
+        for key, value in vars(args).items():
+            if str(value) == "common-ancestor":
+                raise UIError(
+                    f"common-ancestor should not be used argument when bisecting but is used for '{key}'"
+                )
+
+    def run_bisect(
+        self,
+        args: argparse.Namespace,
+        older_commit_hash: str,
+        newer_commit_hash: str,
+        older_commit_success: bool,
+        newer_commit_success: bool,
+        tag_param_instead_checkout: str | None,
+        error_message_must_contain: str | None,
+        log: list[str],
+    ) -> str:
+        commits_in_between = git.get_commits_in_between(
+            older_commit_hash, newer_commit_hash
+        )
+        log.append(
+            f"Commits between {older_commit_hash} and {newer_commit_hash}: {commits_in_between}"
+        )
+
+        if len(commits_in_between) == 0:
+            assert older_commit_success != newer_commit_success
+            return newer_commit_hash
+
+        middle_commit_hash = self._pick_middle_commit(commits_in_between)
+
+        middle_commit_success = self._run_revision(
+            args,
+            middle_commit_hash,
+            tag_param_instead_checkout,
+            error_message_must_contain,
+            log,
+        )
+
+        if middle_commit_success == older_commit_success:
+            older_commit_hash = newer_commit_hash
+        else:
+            newer_commit_hash = middle_commit_hash
+
+        return self.run_bisect(
+            args,
+            older_commit_hash=older_commit_hash,
+            newer_commit_hash=newer_commit_hash,
+            older_commit_success=older_commit_success,
+            newer_commit_success=newer_commit_success,
+            tag_param_instead_checkout=tag_param_instead_checkout,
+            error_message_must_contain=error_message_must_contain,
+            log=log,
+        )
+
+    def handle_result_after_test_run(self, composition: Composition, find: str) -> None:
+        # TODO: extract error infos from test results
+        composition.test_results.clear()
+
+    def _pick_middle_commit(self, commits: list[str]) -> str:
```

Comment:
`git bisect --first-parent` seems better, it will reduce the number of builds required.

Reply:
This would operate on a minimum amount of commits because the commit list is already prefiltered with `--first-parent`.